esp/src/eclwatch/DynamicESDLQueryWidget.js

```diff
@@ -162,6 +162,10 @@ define([
 
                 context.store.setData(results);
                 context.grid.set("query", {__hpcc_parentName: null });
+
+                if (!results.length) {
+                    context.grid.noDataNode.innerHTML = "<span class='dojoxGridNoData' style=\"padding:10px 0 0 5px;\">" + context.i18n.DynamicNoServicesFound + "</span>";
```

Comment:
I think the correct way to set the "no data message" is at Grid creation time (example from web):
```
grid = new Grid({
    store: observableJsonStore,
    queryOptions: queryOptions,
    columns: columns,
    loadingMessage: "Loading data...",
    noDataMessage: "No results found."
}, "grid");
```

Reply:
(from memory hard setting it like that may mean it doesn't get cleared out 100% when data becomes available).

Reply:
Wouldn't this override the functionality of what GridDetailsWidget do for us?

Reply:
It would be more like:
```
            var retVal = new declare([ESPUtil.Grid(false, true)])({
                store: this.store,
                columns: {
                    Name: tree({
                        collapseOnRefresh: false, label: this.i18n.Name, sortable: true, width:200
                    })
                },
                loadingMessage: "Loading data...",
                noDataMessage: "No results found."
            }, domID);
```